internals/overlord/servstate/handlers.go

```diff
@@ -338,15 +358,22 @@ func (s *serviceData) startInternal() error {
 	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
 	// Copy environment to avoid updating original.
-	environment := make(map[string]string)
+	environment := make(map[string]string, len(s.config.Environment))
 	for k, v := range s.config.Environment {
 		environment[k] = v
 	}
 
 	s.cmd.Dir = s.config.WorkingDir
 
 	// Start as another user if specified in plan.
-	uid, gid, err := osutil.NormalizeUidGid(s.config.UserID, s.config.GroupID, s.config.User, s.config.Group)
+	var uid, gid *int
+	if s.config.UserID != nil || s.config.GroupID != nil || s.config.User != "" || s.config.Group != "" {
+		// User/group config from the service takes precedence
+		uid, gid, err = osutil.NormalizeUidGid(s.config.UserID, s.config.GroupID, s.config.User, s.config.Group)
+	} else if s.workload != nil {
+		// Take user/group config from workload
+		uid, gid, err = osutil.NormalizeUidGid(s.workload.UserID, s.workload.GroupID, s.workload.User, s.workload.Group)
+	}
```

Comment:
What about the `else` case?

Reply:
`uid`, `gid` are `nil`, and `setCmdCredential()` is not called, resulting in the services being run as the current user

Reply:
Ok, please resolve